code/framework/object/src/main/java/io/cattle/platform/object/meta/impl/DefaultObjectMetaDataManager.java

```diff
@@ -221,14 +226,33 @@ protected void findMappings() {
             Pair<Class<?>, Relationship> left = rels.get(0);
             Pair<Class<?>, Relationship> right = rels.get(1);
 
-            register(entry.getKey(), left, right);
-            register(entry.getKey(), right, left);
+            if (left.getLeft().equals(right.getLeft())) {
+                // This basically looks like a junction map to itself.
+                // Inspect the column names for extra info and use the link override
+                // to obtain a more meaningful link name
+                String mapRightToLeftName = right.getRight().getPropertyName();
```

Comment:
Basically use the DB column name as a hint.  I also ended up reusing the object.link.name override hook so we get a more user friendly name.